### DISCUSSION TITLE
Node running out of Heap Memory

### DISCUSSION THREAD

**Discussion by gerrysweeney** [vuejs/vitepress#1458] [General]
I have been trying to use Vitepress to generate documentation, but it fails on build with a Node.js "Out of Heap Memory" error. It seems to keep everything in memory while building, it never creates the dist folder, consumes over 4Gb of memory and then I get the attached error.  Any idea how I can resolve that? and/or can the build process be made more memory efficient?

<img width="1424" alt="Screenshot 2022-10-10 at 20 47 31" src="https://user-images.githubusercontent.com/9318317/194942287-be08280c-3778-41e6-88a5-0b7a7d2e5315.png">

**Comment by wkande** [comment]
Did you try settings `export NODE_OPTIONS=--max-old-space-size=9728` to a larger amount of memory. The example here uses 9.5 GB

**Comment by gerry-hornbill** [comment]
I am afraid not, I moved on from vitepress now because I could not make it work for our use case, this problem was the trigger, but the use of webpack (or whatever is turning all the content into JavaScript) just gets too big on our project.

**Comment by wkande** [comment]
See this discussion thread
<https://github.com/vuejs/vitepress/discussions/1778> which has a link to a
repo with instructions to reproduce.

Thanks
